onc-healthit/lantern-back-end
# Validation checks

The current validation logic is all in `validation.go` in the capabilityhandler package, since this is a non-permanent solution.

To test:
- `make run`
- At least run: `go run endpointmanager/cmd/capabilityreceiver/main.go`
  - You can also run the chplquerier and the endpointpopulator before running the above command, but it's only necessary to run the capabilityreceiver
- Check at least the validation field in the fhir_endpoint database table. I've been using a grafana table to make it easier to see all of the fields in the table, but you can also check by running `docker exec -it pg_prometheus psql -U lantern` in a terminal.

----

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-143
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] ~Tests are complete.~ n/a - short term code that we will be removing.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] ~Tests are complete.~ n/a - short term code that we will be removing.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.
